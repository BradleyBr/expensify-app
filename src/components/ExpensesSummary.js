import React from 'react'
import { selectExpensesTotal } from '../selectors/expenses-total'
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'

export class ExpensesSummary extends React.Component{
   expenseCount() {
    return this.props.expenses.length
   }
    
    render() {
        return (
            <div>
                {this.expenseCount() === 0 ? <p>Viewing 0 expenses</p> :
                (this.expenseCount() === 1 ? <p>Viewing 1 expense totalling {numeral(selectExpensesTotal(this.props.expenses)/100).format('$0,0.00')}</p>
                :<p>Viewing {this.expenseCount()} expenses totalling {numeral(selectExpensesTotal(this.props.expenses)/100).format('$0,0.00')}</p>) }
            </div>
        
        )
    }
}

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
})

export default connect(mapStateToProps)(ExpensesSummary)

