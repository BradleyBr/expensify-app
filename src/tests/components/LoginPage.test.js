import { shallow } from 'enzyme'
import React from 'react'
import {LoginPage} from '../../components/LoginPage'

test('Should render Login page', () => {
    const wrapper = shallow(<LoginPage startLogin={() => {}}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should call startLogin on button click', () => {
    let startLogin = jest.fn()
    const wrapper = shallow(<LoginPage startLogin={startLogin} />)
    wrapper.find('button').simulate('click')
    expect(startLogin).toHaveBeenCalled()
})