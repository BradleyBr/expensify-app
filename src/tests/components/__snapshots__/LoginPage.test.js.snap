// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Login page 1`] = `
<div>
  <h3>
    Expensify
  </h3>
  <button
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
